lib/LaTeXML/Post/CrossRef.pm

```diff
@@ -631,7 +631,7 @@ sub generateURL {
         $url .= '#' . $fragid; }
       elsif ($location eq $doclocation) {
         $url = ''; }
-      return $url; }
+      return pathname_to_url($url); }
```

Comment:
Big problem here: apparently `pathname_to_url('./') eq ''`. This breaks `--splitnaming=*relative` because certain links are to `./`, they become empty, and the links end up with no `@href`.